pkg/api/v1/maintenancelimit_types.go

```diff
@@ -19,11 +19,18 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// MaintenanceLimitPolicies optional configuration
+type MaintenanceLimitPolicies struct {
+	RespectNotReadyNodes      bool `json:"respectNotReadyNodes,omitempty"`
```

Comment:
or you could just have string instead of enum makes it easier to implement, not a big fan of enums

Reply:
Not quite sure I understand why an enum or string is superior. Are you thinking that some policies will have more options beyond true/false?

Reply:
Thats the case its always true or false, meaning either its present or not, So you can represent this as an array of Policies rather than explicit true or false.

Example:
```
type MaintenanceLimitPolicy string

const (
    Policy1 MaintenanceLimitPolicy = "Policy1"
    Policy2 MaintenanceLimitPolicy = "Policy2"
)

type MaintenanceLimitSpec struct {
    Policies []MaintenanceLimitPolicy
}
```
So its completely future proof, since all you need to do is add a new constant and a handler for that policy.

Obviously you need to add validation as well :)

Also in future if a policy has a configuration you can create a separate option called 
```
PolicyConfiguration:
    Policy2:
        X: Y
```